src/main/java/com/amazon/opendistro/elasticsearch/performanceanalyzer/decisionmaker/deciders/QueueHealthDecider.java

```diff
@@ -121,10 +126,16 @@ private ModifyQueueCapacityAction configureQueueCapacity(NodeKey esNode, Resourc
   }
 
   private int getNodeQueueCapacity(NodeKey esNode, ResourceEnum threadPool) {
-    // TODO: use NodeConfigurationRca to return capacity, for now returning defaults
-    if (threadPool.equals(ResourceEnum.SEARCH_THREADPOOL)) {
-      return 1000;
+    try {
+      return (int) getAppContext().getNodeConfigCache().get(esNode, Resource.newBuilder()
+              .setResourceEnum(threadPool)
+              .setMetricEnum(MetricEnum.QUEUE_CAPACITY).build());
+    } catch (Exception e) {
+      LOG.error("Exception while reading values from Node Config Cache", e);
     }
-    return 100;
+    // No action if value not present in the cache.
+    // Assumption here is the cache has been wiped off due to
+    // unforeseen events and we dont want to trigger any action.
+    return -1;
```

Comment:
Minor: This is bit of a code smell. Can we raise an exception here instead and handle it gracefully at the caller? Instead of overloading the meaning of return value?

Reply:
Done